File: 2021/js/Day-01/Day-01.js
const fs = require("fs");

inputData = fs.readFileSync("input.txt", "utf8");
inputData = inputData.split(/\r?\n/).map((v) => Number(v));

const Solution01 = (inputData) => {
  let counter = 0;

  for (let i = 0; i < inputData.length; i++) {
    if (i === 0) continue;
    if (inputData[i] > inputData[i - 1]) counter += 1;
  }

  return counter;
};

const Solution02 = (inputData) => {
  let counter = 0;
  let windowSums = [];

  //Create an array with the sum of the moving windows
  for (let i = 0; i < inputData.length - 2; i++) {
    const windowSum = inputData.slice(i, i + 3).reduce((sum, v) => sum + v);
    windowSums.push(windowSum);
  }

  //check how many times the moving windows increase
  for (let i = 0; i < windowSums.length; i++) {
    if (i === 0) continue;
    if (windowSums[i] > windowSums[i - 1]) counter += 1;
  }

  return counter;
};

console.log(`Solution 01: ${Solution01(inputData)}`);
console.log(`Solution 02: ${Solution02(inputData)}`);

File: 2021/python/Day-01/Day_01.py
import time

with open("input.txt","r") as f:
	input_list = [int(i.strip()) for i in f.readlines()]

def solution01(input_list):
	increase_counter = 0 
	for i, _ in enumerate(input_list):
		if i == 0:
			continue

		if input_list[i] > input_list[i-1]:
			increase_counter += 1

	return increase_counter

def solution02(input_list):
	windows_size = 3
	cleaned_data = []

	for i, _ in enumerate(input_list):
		window = input_list[i:i+windows_size]

		if len(window) < windows_size:
			break

		cleaned_data.append(sum(window))

	increase_counter = 0
	for i, _ in enumerate(cleaned_data):
		if i == 0:
			continue

		if cleaned_data[i] > cleaned_data[i-1]:
			increase_counter += 1

	return increase_counter


solution01_start_time = time.time()
solution01 = solution01(input_list)
solution01_total_time = time.time() - solution01_start_time
print(f"Solution 01: {solution01}, in {solution01_total_time}")

solution02_start_time = time.time()
solution02 = solution02(input_list)
solution02_total_time = time.time() - solution02_start_time
print(f"Solution 02: {solution02}, in {solution02_total_time}")

